x/evm/genesis.go

```diff
@@ -39,30 +38,31 @@ func InitGenesis(
 	for _, account := range data.Accounts {
 		address := common.HexToAddress(account.Address)
 		accAddress := sdk.AccAddress(address.Bytes())
+
 		// check that the EVM balance the matches the account balance
 		acc := accountKeeper.GetAccount(ctx, accAddress)
 		if acc == nil {
 			panic(fmt.Errorf("account not found for address %s", account.Address))
 		}
 
-		ethAcct, ok := acc.(evmostypes.EthAccountI)
-		if !ok {
-			panic(
-				fmt.Errorf("account %s must be an EthAccount interface, got %T",
-					account.Address, acc,
-				),
-			)
-		}
 		code := common.Hex2Bytes(account.Code)
 		codeHash := crypto.Keccak256Hash(code)
 
-		// we ignore the empty Code hash checking, see ethermint PR#1234
-		if len(account.Code) != 0 && !bytes.Equal(ethAcct.GetCodeHash().Bytes(), codeHash.Bytes()) {
-			s := "the evm state code doesn't match with the codehash\n"
-			panic(fmt.Sprintf("%s account: %s , evm state codehash: %v, ethAccount codehash: %v, evm state code: %s\n",
-				s, account.Address, codeHash, ethAcct.GetCodeHash(), account.Code))
-		}
-
+		// TODO: I think this can be removed now that the EVM state is detached from the account keeper state
+		// This basically was cross-checking that both are in sync.
+		//
+		// // we ignore the empty Code hash checking, see ethermint PR#1234
+		// if len(account.Code) != 0 && !bytes.Equal(ethAcct.GetCodeHash().Bytes(), codeHash.Bytes()) {
+		// 	s := "the evm state code doesn't match with the codehash\n"
+		// 	panic(fmt.Sprintf("%s account: %s , evm state codehash: %v, ethAccount codehash: %v, evm state code: %s\n",
+		// 		s, account.Address, codeHash, ethAcct.GetCodeHash(), account.Code))
+		// }
```

Comment:
I think this can be removed because it was cross-checking the code hash from the `EthAccount` in the account keeper with the resulting code hash from the actual code loaded during the genesis - can I get some thumbs up/down if that's correct? Do we need alternative ways of checking integrity here? 🤔

Reply:
yes, makes sense, previously the code hash was duplicated. Now there's only one source of truth